python/ffsim/operators/fermi_hubbard.py

```diff
@@ -0,0 +1,81 @@
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Fermi-Hubbard model Hamiltonian."""
+
+from ffsim._lib import FermionOperator
+from ffsim.operators.fermion_action import cre_a, cre_b, des_a, des_b
+
+
+def fermi_hubbard_1d(
+    norb: int,
+    tunneling: float,
+    interaction: float,
+    chemical_potential: float = 0,
+    nearest_neighbor_interaction: float = 0,
+    periodic: bool = False,
+) -> FermionOperator:
+    r"""1D Fermi-Hubbard model Hamiltonian.
+
+    The Hamiltonian for the 1D Fermi-Hubbard model with :math:`N` spatial orbitals is
+    given by
+
+    .. math::
+
+        H = -t \sum_{p=1}^{N-1} \sum_{\sigma}
+        (a^\dagger_{p+1, \sigma} a_{p, \sigma} + \text{H.c.})
+        + \frac{U}{2} \sum_{p=1}^{N} \sum_{\sigma, \sigma'}
+        a^\dagger_{p, \sigma} a^\dagger_{p, \sigma'} a_{p, \sigma'} a_{p, \sigma}
+        - \mu \sum_{p=1}^N \sum_{\sigma} a^\dagger_{\sigma, p} a_{\sigma, p}
+        + V \sum_{p=1}^{N-1} \sum_{\sigma, \sigma'}
+        a^\dagger_{p, \sigma} a^\dagger_{p+1, \sigma'} a_{p+1, \sigma'} a_{p, \sigma}
```

Comment:
Please rewrite the interaction terms by defining the number operator
```
n_{\sigma, p} = a_{\sigma, p}^\dagger a_{\sigma, p}
```
I think that will make it a bit more readable. Once you do that, make sure the order of the operators in the code matches the order in the documentation. Also, please consistently use (spin, index), i.e. `\sigma, p` instead of `p, \sigma`.

Also, I know the Hubbard model is very common, but it would still be good to have a reference for it.